src/NSwag.AspNetCore/Middlewares/SwaggerDocumentKey.cs

```diff
@@ -0,0 +1,35 @@
+namespace NSwag.AspNetCore.Middlewares
+{
+    internal class SwaggerDocumentKey
```

Comment:
Can't we just use `Tuple<string, string, int>` (basePath, host, port) to avoid this code?

Reply:
What about named tuples? Of course, it'll require adding 
`<PackageReference Include="System.ValueTuple" Version="4.5.0" />`
but having something like
`(string basePath, string host, int port)` makes the intention behind tuple's arguments explicit.

The other thing is that we store the tuples in a dictionary and tuples are value types, so it'll add some boxing overhead, however, I don't think the swagger API is in the hot path, so I shouldn't be an issue.